soh/soh/Enhancements/AssignableTunicsAndBoots.cpp

```diff
@@ -94,10 +94,19 @@ void RegisterAssignableTunicsBoots() {
             return;
         }
 
-        Input* input = va_arg(args, Input*);
-        Player* player = GET_PLAYER(gPlayState);
+        Player* player = va_arg(args, Player*);
```

Comment:
Does this make a difference?

Reply:
yes. this is what caused the bug in the first place. `GET_PLAYER` always gets link

Reply:
to elaborate on this:

we make it in here as both link and dark link. the previous logic (before moving this to shipinit) had both link and dark link running both the actionfunc and usetunicboots. by using `GET_PLAYER`, we were running both the actionfunc and usetunicboots *as link* for both link and dark link.

so we need to know if we're in here because of link or dark link

i could switch to passing in the actor id instead and then using `GET_PLAYER` if you'd prefer that

Reply:
Nope that's fine, I just misunderstood what was happening.

Reply:
Yeah we had to do something similar in a 2ship hook because of player common being used by Kafei and Link.